metric/handler.go

```diff
@@ -26,13 +27,15 @@ var errAccessDenied = errors.New("Access denied")
 type Server struct {
 	metrics http.Handler
 	session core.Session
+	config  config.Config
 }
 
 // NewServer returns a new metrics server.
-func NewServer(session core.Session) *Server {
+func NewServer(session core.Session, config config.Config) *Server {
```

Comment:
we typically try to avoid passing the entire config into different packages. So in this case I recommend the following for consistency with our other packages:

```
-func NewServer(session core.Session, config config.Config)
+func NewServer(session core.Session, anonymous bool)
```

Reply:
note that this will require modifying https://github.com/drone/drone/blob/master/cmd/drone-server/inject_server.go and creating a provider function. For example:

```diff
// wire set for loading the server.
var serverSet = wire.NewSet(
	manager.New,
-	metric.NewServer,
	api.New,
	web.New,
+	provideMetrics,
	provideRouter,
	provideRPC,
	provideServer,
	provideServerOptions,
)
```